### PR TITLE
Files for nucleo-h743zi2 and pysimCoder

### PR DESCRIPTION
**Pull request by robertobucher**
## Summary
These files are required to use the nucleo-h743ZI2 board with pysimCoder

The following features are possible:
5 ADC
4 DI
8 DO (including 3 LED)
2 PWM channels with complementary signals
2 encoders

## Impact
arch/arm/src/stm32h7/stm32_adc.c (output formats)
arch/arm/src/stm32h7/stm32_qencoder.c (output formats)
board/qrm/stm32h7/nucleo-h743zi2/include/board.h (timers for PWM and encoders)
board/qrm/stm32h7/nucleo-h743zi2/src/Makefile (additional files)
board/qrm/stm32h7/nucleo-h743zi2/src/nucleo-h743zi2.h (gpio)
board/qrm/stm32h7/nucleo-h743zi2/src/stm32_adc.c (new)
board/qrm/stm32h7/nucleo-h743zi2/src/stm32_bringup.c
board/qrm/stm32h7/nucleo-h743zi2/src/stm32_gpio.c (new)
board/qrm/stm32h7/nucleo-h743zi2/src/stm32_pwm.c (new)
board/qrm/stm32h7/nucleo-h743zi2/src/stm32_qencoder.c (new)

## Testing
Tested using pysimCoder on nucleo-H743ZI2 board

### CODE REVIEW

File: boards/arm/stm32h7/nucleo-h743zi2/include/board.h [hunk 1]
```diff
@@ -395,7 +395,7 @@
 
 /* TIM3 */
 
-#define GPIO_TIM3_CH1OUT  GPIO_TIM3_CH1OUT_2    /* PB4  */
+#define GPIO_TIM3_CH1OUT  GPIO_TIM3_CH1OUT_2   /* PB4  */
```
Comment (xiaoxiang781216) [inline boards/arm/stm32h7/nucleo-h743zi2/include/board.h:398]: squash to the previous patch

**Comment by xiaoxiang781216** [comment]
@robertobucher it's better to squash the last 5 patch into one.

File: arch/arm/src/stm32h7/stm32_adc.c [hunk 2]
```diff
@@ -1478,17 +1478,19 @@ static int adc_setup(struct adc_dev_s *dev)
...
-  ainfo("ISR:   0x%08x CR:    0x%08x CFGR:  0x%08x CFGR2: 0x%08x\n",
+  ainfo("ISR:   0x%08" PRIx32 " CR:    0x%08" PRIx32 " \
+          CFGR:  0x%08" PRIx32 " CFGR2: 0x%08" PRIx32 "\n",
         adc_getreg(priv, STM32_ADC_ISR_OFFSET),
         adc_getreg(priv, STM32_ADC_CR_OFFSET),
         adc_getreg(priv, STM32_ADC_CFGR_OFFSET),
         adc_getreg(priv, STM32_ADC_CFGR2_OFFSET));
-  ainfo("SQR1:  0x%08x SQR2:  0x%08x SQR3:  0x%08x SQR4:  0x%08x\n",
+  ainfo("SQR1:  0x%08" PRIx32 " SQR2:  0x%08" PRIx32 " \
+        SQR3:  0x%08" PRIx32 " SQR4:  0x%08" PRIx32 "\n",
```
Comment (pkarashchenko) [inline arch/arm/src/stm32h7/stm32_adc.c:1488]:
```suggestion
  ainfo("SQR1:  0x%08" PRIx32 " SQR2:  0x%08" PRIx32 " "
        "SQR3:  0x%08" PRIx32 " SQR4:  0x%08" PRIx32 "\n",
```

File: arch/arm/src/stm32h7/stm32_adc.c [hunk 3]
```diff
@@ -1854,8 +1856,8 @@ static int adc_interrupt(struct adc_dev_s *dev, uint32_t adcisr)
       value  = adc_getreg(priv, STM32_ADC_DR_OFFSET);
       value &= ADC_DR_MASK;
 
-      awarn("WARNING: Analog Watchdog, Value (0x%03x) out of range!\n",
-            value);
+      awarn("WARNING: Analog Watchdog, Value (0x%03" PRIx32 ") \
+              out of range!\n", value);
```
Comment (pkarashchenko) [inline arch/arm/src/stm32h7/stm32_adc.c:1860]:
```suggestion
      awarn("WARNING: Analog Watchdog, Value (0x%03" PRIx32 ") "
            "out of range!\n", value);
```

File: arch/arm/src/stm32h7/stm32_qencoder.c [hunk 4]
```diff
@@ -987,7 +987,8 @@ static int stm32_shutdown(struct qe_lowerhalf_s *lower)
   putreg32(regval, regaddr);
   leave_critical_section(flags);
 
-  sninfo("regaddr: %08x resetbit: %08x\n", regaddr, resetbit);
+  sninfo("regaddr: %08" PRIx32 " resetbit: %08" PRIx32 "\n", \
+          regaddr, resetbit);
```
Comment (pkarashchenko) [inline arch/arm/src/stm32h7/stm32_qencoder.c:991]:
```suggestion
  sninfo("regaddr: %08" PRIx32 " resetbit: %08" PRIx32 "\n",
         regaddr, resetbit);
```

File: boards/arm/stm32h7/nucleo-h743zi2/src/nucleo-h743zi2.h [hunk 5]
```diff
@@ -101,6 +101,39 @@
...
+#define GPIO_OUT5         (GPIO_OUTPUT | GPIO_PUSHPULL | GPIO_SPEED_50MHz | \
+                           GPIO_OUTPUT_SET | GPIO_PORTF | GPIO_PIN3)
+#define GPIO_INT1         (GPIO_INPUT | GPIO_FLOAT | GPIO_PORTE | GPIO_PIN3)
+
+/* PWM */
+
+#if defined(CONFIG_STM32H7_TIM1_PWM)
+# define NUCLEOH743ZI2_PWMTIMER 1
+#else
+# define NUCLEOH743ZI2_PWMTIMER 3
```
Comment (pkarashchenko) [inline boards/arm/stm32h7/nucleo-h743zi2/src/nucleo-h743zi2.h:134]:
```suggestion
#  define NUCLEOH743ZI2_PWMTIMER 1
#else
#  define NUCLEOH743ZI2_PWMTIMER 3
```

File: boards/arm/stm32h7/nucleo-h743zi2/src/nucleo-h743zi2.h [hunk 6]
```diff
@@ -148,4 +205,20 @@ void weak_function stm32_usbinitialize(void);
...
+ *   Initialize PWM and register the PWM device.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_PWM
+int stm32_pwm_setup(void);
+#endif
+
+#ifdef CONFIG_SENSORS_QENCODER
+int stm32_qencoder_initialize(const char *devpath, int timer);
```
Comment (pkarashchenko) [inline boards/arm/stm32h7/nucleo-h743zi2/src/nucleo-h743zi2.h:221]: Please add function description

File: boards/arm/stm32h7/nucleo-h743zi2/src/stm32_pwm.c [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+  int ret;
+
+  /* Have we already initialized? */
+
+  if (!initialized)
+    {
+      /* Get an instance of the PWM interface */
+
+      pwm = stm32_pwminitialize(NUCLEOH743ZI2_PWMTIMER);
+      if (!pwm)
```
Comment (pkarashchenko) [inline boards/arm/stm32h7/nucleo-h743zi2/src/stm32_pwm.c:77]:
```suggestion
      if (pwm == NULL)
```

**Review by pkarashchenko — COMMENTED** [review]
In general looks good

**Pull request closed without merging**